src/coverlet.console/Program.cs

```diff
@@ -223,7 +225,16 @@ static int Main(string[] args)
                         exceptionMessageBuilder.AppendLine($"The {dThresholdStat.ToString().ToLower()} method coverage is below the specified {dThreshold}");
                     }
 
-                    throw new Exception(exceptionMessageBuilder.ToString());
+                    switch (dThresholdAct)
+                    {
+                        case ThresholdAction.Warning:
+                            logger.LogWarning(exceptionMessageBuilder.ToString());
+                            break;
+                        case ThresholdAction.Fail:
+                        default:
+                            exitCode += (int)CommandExitCodes.CoverageBelowThreshold;
```

Comment:
I prefer a default with an error like `throw new ArgumentException("Unhandled ThresholdAction")` or something like that, than default `Fail` if we add a new enum and we don't handle it we should catch error immediatly, some test could get false positive(fails but not for right reason).